oidc.go

```diff
@@ -238,6 +242,73 @@ func (p *Provider) UserInfo(ctx context.Context, tokenSource oauth2.TokenSource)
 	return &userInfo, nil
 }
 
+// EndSessionOptions is optional parameters for the Initiated Logout by https://openid.net/specs/openid-connect-session-1_0.html#RPLogout
+type EndSessionOptions struct {
+	IDTokenHint string
+	RedirectURL string
+	State       string
+}
+
+// EndSession uses the token source to query the provider's end session endpoint.
+func (p *Provider) EndSession(ctx context.Context, tokenSource oauth2.TokenSource, options EndSessionOptions) error {
+	endSessionURL, err := makeEndSessionEndpoint(p.endSessionURL, options)
+	if err != nil {
+		return fmt.Errorf("oidc: error of make an end session url: %v", err)
+	}
+
+	req, err := http.NewRequest(http.MethodGet, endSessionURL, nil)
+	if err != nil {
+		return fmt.Errorf("oidc: create GET request: %v", err)
+	}
+
+	token, err := tokenSource.Token()
+	if err != nil {
+		return fmt.Errorf("oidc: get access token: %v", err)
+	}
+
+	token.SetAuthHeader(req)
+
+	resp, err := doRequest(ctx, req)
```

Comment:
I'm not sure it's correct for the library to make this request. The spec specifically calls out that this should be initiated by the browser, not the server

```
An RP can notify the OP that the End-User has logged out of the site, and might want to log out of the OP as well. In this case, the RP, after having logged the End-User out of the RP, redirects the End-User's User Agent to the OP's logout endpoint URL. This URL is normally obtained via the end_session_endpoint element of the OP's Discovery response, or may be learned via other mechanisms.
```
https://openid.net/specs/openid-connect-session-1_0-17.html#RPLogout